client/components/filters/compare/index.js

```diff
@@ -31,12 +32,23 @@ class CompareFilter extends Component {
 		}
 	}
 
-	componentDidUpdate( { param } ) {
-		if ( param !== this.props.param ) {
+	componentDidUpdate( { param: prevParam, query: prevQuery } ) {
+		const { getLabels, param, path, query } = this.props;
+		if ( prevParam !== param ) {
 			/* eslint-disable react/no-did-update-set-state */
 			this.setState( { selected: [] } );
-			updateQueryString( { [ param ]: '' }, this.props.path, this.props.query );
 			/* eslint-enable react/no-did-update-set-state */
+			updateQueryString( { [ param ]: '' }, path, query );
+			return;
+		}
+
+		const prevIds = getIdsFromQuery( prevQuery[ param ] );
+		const currentIds = getIdsFromQuery( query[ param ] );
+		if ( ! isEqual( prevIds.sort(), currentIds.sort() ) ) {
```

Comment:
Can you compare `state.selected` against `currentIds` to see if they are the same and return without setting state? The reason is manipulating just the Compare card causes the tags to be fetched and disappear